### ISSUE TITLE
Dose not installed latest version

### ISSUE THREAD

**Issue by LisonFan** [CocoaPods/CocoaPods#8697]
# Report

## What did you do?
Run `pod install` and `pod update`

## What did you expect to happen?

Install all pod dependencies correctly.

## What happened instead?

ProjectA dependencies PodA and PodB
PodA has relied on PodB, Includes other dependencies in PodB, but PodA and PodB have no cross-references.
Execute `pod install` or `pod update`, PodA download is the old version 

### Stack

```
   CocoaPods : 1.6.1
        Ruby : ruby 2.6.1p33 (2019-01-30 revision 66950) [x86_64-darwin18]
    RubyGems : 3.0.3
        Host : Mac OS X 10.14.4 (18E226)
       Xcode : 10.2 (10E125)
         Git : git version 2.21.0
Ruby lib dir : /Users/USERNAME/.rvm/rubies/ruby-2.6.1/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ b4733bdd3e1792dd7966814e8a9dfc465bb5d958
               **** - https://git.oschina.net/***/***.git @ ***
               **** - https://****/***/***.git @ ***
```

### Installation Source

```
Executable Path: /Users/USERNAME/.rvm/rubies/ruby-2.6.1/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.4
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.1.0
cocoapods-trunk       : 1.3.1
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
source 'https://github.com/CocoaPods/Specs.git'
source 'https://****/***/***.git'

platform :ios, '9.0'

target '***' do
    pod 'PodA'
    pod 'PodB'
    # pod '***'
    # pod '***'
    # pod '***'
    # pod '***'    
    # pod '***'
    pod 'Masonry'
    pod 'ReactiveObjC'
end

post_install do |installer|
    installer.pods_project.build_configuration_list.build_configurations.each do |configuration|
        configuration.build_settings['CLANG_ALLOW_NON_MODULAR_INCLUDES_IN_FRAMEWORK_MODULES'] = 'YES'
    end
    installer.pods_project.targets.each do |target|
        target.build_configurations.each do |config|
            config.build_settings['ENABLE_BITCODE'] = 'NO'
            config.build_settings['APPLICATION_EXTENSION_API_ONLY'] = 'NO'
        end
    end
end
```

**Comment by segiddins** [comment]
Please share a Podfile that can be used to reproduce the issue

**Comment by LisonFan** [comment]
> Please share a Podfile that can be used to reproduce the issue

@seaders Hello
Because the relevant library is in the company intranet, So I can't give a podfile
But I tried to simulate this situation on the public network and failed.
I tried to execute `export MOLINILLO_DEBUG=1` before `pod install` or `pod update`, Printed log file didn't see it because of what caused PodA to download the old version.
Is there any other way to print detailed logs?
Thank you!
